### DISCUSSION TITLE
Adds query param when using route param with custom request class

### DISCUSSION THREAD

**Discussion by paulchill** [dedoc/scramble#393] [Q&A] [answered]
I have a route which accepts an id as a route param:

`Route::get('process/{id}', [ProcessController::class, 'show']);`

Used in my controller
`public function show(ProcessRequest $request, string $id)`
  
Using a custom Request class

```
use Illuminate\Foundation\Http\FormRequest;
class ProcessRequest extends FormRequest

```
Scramble, adds an an extra id as a query param which i can't seem to remove i use the standard request class

![image](https://github.com/dedoc/scramble/assets/6987900/289621e5-d053-46c9-a60a-e9998ffe8a48)

**Comment by romalytvynenko** [comment]
@paulchill can you share the entire source of your form request class?

**Reply by paulchill** [reply]
Sure!

```
<?php

namespace App\Http\Requests\Models;

use Illuminate\Foundation\Http\FormRequest;

class ProcessRequest extends FormRequest
{
    /**
     * Determine if the user is authorized to make this request.
     *
     * @return bool
     */
    public function authorize()
    {
        return true;
    }

    /**
     * Get the validation rules that apply to the request.
     *
     * @return array
     */
    public function rules()
    {
        return [
            'id' => 'required|string'
        ];
    }

    /**
     * Get the error messages for the defined validation rules.
     *
     * @return array
     */
    public function messages()
    {
        return [
            'id.required' => 'The id field is required.',
            'id.string' => 'The id field must be a string.',
        ];
    }

    /**
     * Include route parameters in validation data.
     *
     * @return array
     */
    public function validationData()
    {
        return array_merge($this->request->all(), [
            'id' => $this->route('id'),
        ]);
    }
}

```

I tried adding in validationData() to see if it would help but no luck.

**Reply by romalytvynenko** [reply]
Oh, I see. Yeah, Scramble indeed "sees" it as a separate parameter as it is presented in the validation rules. 

Will fix this

**Reply by paulchill** [reply]
Great thanks - was driving me nuts :)

**Reply by romalytvynenko** [reply, marked as answer]
@paulchill after more thoughts, I believe that this behavior is correct 🫣 At the same time, the stuff should not drive you nuts! So I'm about to allow to manually opt out of some parameters documentation by using `@ignoreParam` annotation.

So you'd need to write `rules` like so:
```php
public function rules()
{
    return [
        /** @ignoreParam */
        'id' => 'required|string'
    ];
}
```

Does that make sense?

**Reply by paulchill** [reply]
I can't push it to production with the issue :( 

```
 public function rules()
    {
        
        return [
            /** @ignoreParam */
            'id' => 'required|string'
        ];
    }
```

Doesn't effect it - after clearing cache too.
![image](https://github.com/dedoc/scramble/assets/6987900/73fdd120-a4b4-4eb5-81e4-ce9831f82909)

**Reply by romalytvynenko** [reply]
@paulchill yeah yeah, that was just the idea when I wrote the comment. Just released it. Please upgrade to `v0.10.11`

**Reply by paulchill** [reply]
Gent - thank you :)

**Reply by m-macdev** [reply]
Great, the solution works.

I went one more level up.

I am using same request file for different methods. like this

in my BaseRequest file I have 

```
public function rules(): array
    {
        return match ($this->method()) {
            'PUT' => $this->updateValidate(),
            'POST' => $this->storeValidate(),
            'GET' => $this->getValidate(),
            'DELETE' => $this->deleteValidate(),
            'PATCH' => $this->patchValidate(),
            default => [],
        };
    }
```

And In my child request files I have implemented these methods.

so here in one of the files, /** @ignoreParam */, does not work. 

```
public function patchValidate(): array
    {
        return [
            /** @ignoreParam */
            'id' => ['required', 'integer'],
           ...
        ];
    }

```

Is there any solution for this? or I must need to use rule() method?

**Reply by romalytvynenko** [reply]
@m-macdev when diverging from Laravel conventions, you'd need to make your own extensions for such implementations
